### PR TITLE
add webhook config to Terminator

### PR DESCRIPTION
**Pull request by cjerad**
**Issue #, if available:**

#556 Initial NTH v2 design proposal

**Description of changes:**

Add `webhook` property to `Terminator` spec

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: src/pkg/terminator/eventkind.go [hunk 1]
```diff
@@ -16,7 +16,7 @@ limitations under the License.
 
 package terminator
 
-type EventKind string
+type EventKind = string
```
Comment (snay2) [inline src/pkg/terminator/eventkind.go:19]: Can you describe the semantic difference between these two ways of describing the type? I'm not very familiar with it.

File: src/pkg/webhook/builder.go [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+
+type (
+	HttpSendFunc = func(*http.Request) (*http.Response, error)
+	ProxyFunc    = func(*http.Request) (*url.URL, error)
+
+	ClientBuilder func(ProxyFunc) HttpSendFunc
+)
+
+func NewHttpClientDo(proxy ProxyFunc) HttpSendFunc {
+	c := &http.Client{
```
Comment (snay2) [inline src/pkg/webhook/builder.go:36]: Does the HTTP client need any retry or backoff configurations, or is this sufficient for how it makes requests?

**Review by snay2 — APPROVED** [review]

File: src/pkg/terminator/eventkind.go (hunk 1 above)
Comment (cjerad) [reply src/pkg/terminator/eventkind.go:19]:
```go
type EventKind string // Defines a new type whose content is a `string`.
type EventKind = string // Defines an alias for the `string` type, i.e. `EventKind` and `string` values can be used interchangeably.
```

The switch made things a easier when including the event kind description in the webhook notification.

File: src/pkg/webhook/builder.go (hunk 2 above)
Comment (cjerad) [reply src/pkg/webhook/builder.go:36]: This is how NTHv1 configures its client (see [here](https://github.com/aws/aws-node-termination-handler/blob/5e4984f2d8588ce6661cc33ccb836249a85852a8/pkg/webhook/webhook.go#L93-L108)).  Might be worth adding retry, etc. support if we run into problems.

**Pull request merged**